includes/class-monk-links.php

```diff
@@ -624,6 +624,9 @@ public function monk_canonical_redirection() {
 			$redirect_url  = ( ! redirect_canonical( $_redirect_url, false ) ) ? $_redirect_url : redirect_canonical( $_redirect_url, false );
 
 			$redirect_url = $this->monk_change_language_url( $redirect_url, $slug );
+			if ( $this->monk_using_permalinks() ) {
+				$redirect_url = remove_query_arg( 'lang', $redirect_url );
```

Comment:
We shouldn't remove it.
When permalinks are being used, we need support `lang` as a normal parameter without affecting the site language and queries.